### PR TITLE
Add b64url support

### PR DESCRIPTION
**Pull request by dpetran**

### CODE REVIEW

File: src/alphabase/base64url.cljc [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+(def ^:const base64url-chars "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_.")
+(def ^:const base64url-set (set base64url-chars))
+
+(defn encode
+  "Converts a byte array into a base65URL string."
+  ^String
+  [ba]
+  #?(:clj (-> (Base64/getUrlEncoder)
+              (.encodeToString ba))
```
Comment (zonotope) [inline src/alphabase/base64url.cljc:17]: this is so minor it probably doesn't matter, but i think executing this function and `decode` below will result in some reflection on the url encoder object.

**Review by zonotope — APPROVED** [review]
🧮

File: src/alphabase/base64url.cljc (hunk 1 above)
Comment (dpetran) [reply src/alphabase/base64url.cljc:17]: Good thought! I set *warn-on-reflection* true in the `core` ns and fixed all the warnings - it didn't flag this but it did find several others.

**Comment by dpetran** [comment]
@zonotope I added a few commits after you approved

File: src/alphabase/base64url.cljc [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            #?@(:cljs [[goog.crypt.base64 :as g-base64]]))
+  #?(:clj (:import (java.util Base64))))
+
+#?(:clj (set! *warn-on-reflection* true))
+
+(def ^:const base64url-chars "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_.")
+(def ^:const base64url-set (set base64url-chars))
+
+(defn encode
+  "Converts a byte array into a base65URL string."
```
Comment (cap10morgan) [inline src/alphabase/base64url.cljc:13]: s/base65/base64/ :)

Comment (dpetran) [reply src/alphabase/base64url.cljc:13]: :facepalm: I'm just going to push this fix straight to master

Comment (dpetran) [reply src/alphabase/base64url.cljc:13]: https://github.com/fluree/alphabase/commit/1161efe7f5fd5abb066adf2283f5da3c7683a48f

**Pull request merged**